sdk/core/azure_core/src/http/policies/bearer_token_policy.rs

```diff
@@ -30,23 +36,28 @@ impl BearerTokenAuthorizationPolicy {
         A: IntoIterator<Item = B>,
         B: Into<String>,
     {
+        let scopes: Vec<String> = scopes.into_iter().map(|s| s.into()).collect();
         Self {
-            credential,
-            scopes: scopes.into_iter().map(|s| s.into()).collect(),
-            access_token: Arc::new(RwLock::new(None)),
+            authorizer: Arc::new(BearerTokenAuthorizer::new(credential)),
+            on_request: Arc::new(DefaultOnRequest { scopes }),
+            on_challenge: None,
         }
     }
 
-    fn scopes(&self) -> Vec<&str> {
-        self.scopes
-            .iter()
-            .map(String::as_str)
-            .collect::<Vec<&str>>()
+    /// Sets a callback for `send` to invoke once on each request it receives, before sending the request. See [`OnRequest`]
+    /// for more details. When not set, the policy authorizes each request using the credential and scopes specified to `new`.
```

Comment:
Doc comments in Rust are like (good) git commits: first line is a summary, blank line, more info. So this should probably be:


```suggestion
    /// Sets a callback for `send` to invoke once on each request it receives, before sending the request.
    ///
    /// See [`OnRequest`] for more details. When not set, the policy authorizes each request using the credential and scopes specified to `new`.
```